src/printing.jl

```diff
@@ -0,0 +1,29 @@
+
+# Only when AbstractAlgebra.expressify(a::T; context = nothing) has been
+# defined may these be used.
+# AA defines Base.show for "text/latex" and "text/html" for a general set
+# of x, but for backward compatibility it is not defined for general x and
+# "text/plain" or the mime-less version.
+# Rationale: when neither Base.show nor AA.expressify is defined for T, then,
+# since expressify calls Base.show for backward compatibility, a definition of
+# Base.show in terms of expressify would give a stack overflow.
+
+macro enable_all_show_via_expressify(T)
+  return quote
+    function Base.show(io::IO, x::$T)
+       AbstractAlgebra.show_via_expressify(io, x)
+    end
+
+    function Base.show(io::IO, mi::MIME"text/plain", x::$T)
+       AbstractAlgebra.show_via_expressify(io, mi, x)
+    end
+
+    function Base.show(io::IO, mi::MIME"text/latex", x::$T)
+       AbstractAlgebra.show_via_expressify(io, mi, x)
+    end
+
+    function Base.show(io::IO, mi::MIME"text/html", x::$T)
+       AbstractAlgebra.show_via_expressify(io, mi, x)
+    end
+  end
+end
```

Comment:
Should this be in AbstractAlgebra?

Reply:
Maybe hide it behind `isdefined(AbstractAlgebra, Symbol("@enable_all..."))`, then we can move it to AA whenever we want.

Reply:
If you could explain why, when moved to AA, this code needs $(esc(T)), I would be grateful.

Reply:
You should attend the juliacon workshop on metaprogramming.